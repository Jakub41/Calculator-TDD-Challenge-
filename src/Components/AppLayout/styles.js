import styled from "@emotion/styled";

export const ContentLayout = styled.div`
	width: 100px;
	height: 100px;
`
