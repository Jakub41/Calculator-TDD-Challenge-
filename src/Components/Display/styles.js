import styled from "@emotion/styled";

export const DisplayDiv = styled.div`

`;
