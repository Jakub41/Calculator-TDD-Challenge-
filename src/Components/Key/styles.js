import styled from "@emotion/styled";

export const KeyDiv = styled.div`

`;

export const KeyButton = styled.button`

`;

