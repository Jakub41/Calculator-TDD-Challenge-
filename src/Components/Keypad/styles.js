import styled from "@emotion/styled";

export const KeypadDiv = styled.div`

`;

export const NumbersDiv = styled.div`

`;

export const OperatorsDiv = styled.div`

`;
