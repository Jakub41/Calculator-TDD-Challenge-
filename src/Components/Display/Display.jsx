import React from "react";
import PropTypes from "prop-types";
import { DisplayDiv } from "./styles";

const Display = ({ displayValue }) => (
	<>
		<DisplayDiv>{displayValue}</DisplayDiv>
	</>
);

Display.propTypes = { displayValue: PropTypes.string.isRequired };

export default Display;
