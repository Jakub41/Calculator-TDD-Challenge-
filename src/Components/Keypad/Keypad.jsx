import React from "react";
import PropTypes from "prop-types";
import { KeypadDiv, NumbersDiv, OperatorsDiv } from "./styles";

const Keypad = ({
	callOperator,
	numbers,
	operators,
	setOperator,
	updateDisplay,
}) => (
	<KeypadDiv>
		<NumbersDiv></NumbersDiv>
		<OperatorsDiv></OperatorsDiv>
	</KeypadDiv>
);

// Keypad.propTypes = {
// 	callOperator: PropTypes.func.isRequired,
// 	numbers: PropTypes.array.isRequired,
// 	operators: PropTypes.array.isRequired,
// 	setOperator: PropTypes.func.isRequired,
// 	updateDisplay: PropTypes.func.isRequired,
// };

export default Keypad;
