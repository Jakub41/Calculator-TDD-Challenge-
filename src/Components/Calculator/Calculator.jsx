import React, { useState } from "react";
import { CalculatorDiv } from "./styles";
import Display from "../Display";

const Calculator = () => {
	const [displayValue, setDisplayValue] = useState("");

	return (
		<>
			<CalculatorDiv>
				<Display displayValue={displayValue} />
			</CalculatorDiv>
		</>
	);
};

export default Calculator;
