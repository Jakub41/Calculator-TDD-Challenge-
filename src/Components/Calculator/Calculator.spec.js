import React from "react";
import { shallow } from "enzyme";
import Calculator from "./Calculator";
import { CalculatorDiv } from "./styles";
import renderer from "react-test-renderer";
import Display from "../Display";
import Keypad from "../Keypad";

describe("Calculator Component", () => {
	let wrapper;

	beforeEach(() => (wrapper = shallow(<Calculator />)));

	it("Calculator Snapshot renderer", () => {
		const component = renderer.create(<Calculator />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("CalculatorDiv Snapshot renderer", () => {
		const styledComponent = renderer.create(<CalculatorDiv />);
		let tree = styledComponent.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should render Calculator", () => {
		expect(wrapper.find(CalculatorDiv)).toHaveLength(1);
	});

	const displayValue = "0";
	it("should render the Display Component", () => {
		expect(
			wrapper.containsMatchingElement(<Display displayValue={displayValue} />)
		).toEqual(true);
	});

	it("should render the Display and Keypad Components", () => {
		expect(
			wrapper.containsAllMatchingElements([
				<Display displayValue={displayValue} />,
				<Keypad
					// callOperator={wrapper.instance().callOperator}
					// numbers={wrapper.instance().state.numbers}
					// operators={wrapper.instance().state.operators}
					// setOperator={wrapper.instance().setOperator}
					// updateDisplay={wrapper.instance().updateDisplay}
				/>,
			])
		).toEqual(true);
	});
});
