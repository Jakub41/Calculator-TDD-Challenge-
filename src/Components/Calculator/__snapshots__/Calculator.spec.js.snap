// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component Calculator Snapshot renderer 1`] = `
<div
  className="css-1esthja"
>
  <div
    className="css-1esthja"
  >
    0
  </div>
  <div
    className="css-1esthja"
  >
    <div
      className="css-1esthja"
    />
    <div
      className="css-1esthja"
    />
  </div>
</div>
`;

exports[`Calculator Component CalculatorDiv Snapshot renderer 1`] = `
<div
  className="css-1esthja"
/>
`;
